OpenSim/Simulation/Test/testStatesTrajectory.cpp

```diff
@@ -54,12 +54,17 @@ Real getStorageEntry(const Storage& sto,
     return value;
 }
 
-void testPopulateTrajectory() {
+void testPopulateTrajectoryAndStatesCollector() {
     Model model("gait2354_simbody.osim");
 
     // To assist with creating interesting (non-zero) coordinate values:
     model.updCoordinateSet().get("pelvis_ty").setDefaultLocked(true);
 
+    // Also, test the StatesCollector.
+    auto* statesCol = new StatesCollector();
+    statesCol->setName("states_collector");
+    model.addModelComponent(statesCol);
```

Comment:
Adding `statesCol` to the Model doesn't quite seem right to me. Will it eventually be added to a Study?